File: b_rank_skillcheck_sample/long-table_05.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] nm = ReadLine().Split().Select( int.Parse ).ToArray();

        foreach( int _ in Range( 0, nm[1] ) ) {
            int[] ab = ReadLine().Split().Select( int.Parse ).ToArray();
            var l = new System.Collections.Generic.List<int>( ab[0] );

            foreach( int i in Range( ab[1], ab[0] ) ) {
                l.Add( nm[0] < i ? i - nm[0] : i );
            }
            WriteLine( string.Join( " ", l ) );
        }
        return;
    }
}

File: b_rank_skillcheck_sample/long-table_05.py
# 実行時間 10ms
n, m = list(map(int, input().split()))

for _ in [0] * m:
    a, b = list(map(int, input().split()))
    l = []

    for i in range(b, a + b):
        l.append(i - n if n < i else i)
    print(*l)
